clients/src/test/java/org/apache/kafka/clients/consumer/StickyAssignorTest.java

```diff
@@ -272,6 +284,73 @@ public void testSchemaBackwardCompatibility() {
         assertTrue(isFullyBalanced(assignment));
     }
 
+    @Test
+    public void testMemberDataWithInconsistentData() {
+        Map<String, Integer> partitionsPerTopic = new HashMap<>();
+        partitionsPerTopic.put(topic, 2);
+        List<TopicPartition> ownedPartitionsInUserdata = partitions(tp1);
+        List<TopicPartition> ownedPartitionsInSubscription = partitions(tp0);
+
+        assignor.onAssignment(new ConsumerPartitionAssignor.Assignment(ownedPartitionsInUserdata), new ConsumerGroupMetadata(groupId, generationId, consumer1, Optional.empty()));
+        ByteBuffer userDataWithHigherGenerationId = assignor.subscriptionUserData(new HashSet<>(topics(topic)));
+        // The owned partitions and generation id are provided in user data and different owned partition is provided in subscription without generation id
+        // If subscription provides no generation id, we'll honor the generation id in userData and owned partitions in subscription
+        Subscription subscription = new Subscription(topics(topic), userDataWithHigherGenerationId, ownedPartitionsInSubscription);
+
+        AbstractStickyAssignor.MemberData memberData = memberData(subscription);
+
+        // In StickyAssignor, we'll serialize owned partition in assignment into userData
+        assertEquals(ownedPartitionsInUserdata, memberData.partitions, "subscription: " + subscription + " doesn't have expected owned partition");
+        assertEquals(generationId, memberData.generation.orElse(-1), "subscription: " + subscription + " doesn't have expected generation id");
+    }
+
+    @Test
+    public void testMemberDataWithEmptyPartitionsAndEqualGeneration() {
```

Comment:
I wonder if this test is useful with the last version. We always take from user data anyway.

Reply:
Make sense. Removed.